### PR TITLE
mptensor: Add test method

### PR DESCRIPTION
**Pull request by kuramoto-fj**

### CODE REVIEW

**Comment by kuramoto-fj** [comment]
Could you review this PR?
If there is no problem with the recipe on the PR, could you merge it?

File: var/spack/repos/builtin/packages/mptensor/package.py [hunk 1]
```diff
@@ -11,45 +11,95 @@ class Mptensor(CMakePackage):
...
+        blas = spec["blas"].libs
+        lapack = spec["lapack"].libs
+        options.extend(
+            [
+                "-DLAPACK_LIBRARIES=%s" % ";".join(lapack),
+                "-DBLAS_LIBRARIES=%s" % ";".join(blas),
+                self.define_from_variant("ENABLE_MPI", "mpi"),
+                self.define_from_variant("BUILD_DOC", "doc"),
+            ]
+        )
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mptensor/package.py]: These seem just style changes. Can you revert them?

File: var/spack/repos/builtin/packages/mptensor/package.py [hunk 2]
```diff
@@ -11,45 +11,95 @@ class Mptensor(CMakePackage):
...
-                '-DSCALAPACK_LIBRARIES=%s'    % spec['scalapack'].libs,
-            ])
+            options.extend(
+                [
+                    "-DCMAKE_C_COMPILER=%s" % spec["mpi"].mpicc,
+                    "-DCMAKE_CXX_COMPILER=%s" % spec["mpi"].mpicxx,
+                    "-DCMAKE_Fortran_COMPILER=%s" % spec["mpi"].mpifc,
+                    "-DSCALAPACK_LIBRARIES=%s" % spec["scalapack"].libs,
+                ]
+            )
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mptensor/package.py]: Same here

**Review by alalazo — CHANGES_REQUESTED** [review]
Can you revert all the changes that are purely on style? I marked a couple but I think there are more.

**Comment by kuramoto-fj** [comment]
Thank you for your reviews.  And sorry for changed style.
Because I used py-black, so the style has changed.
I reverted the receipe on merged style.

File: var/spack/repos/builtin/packages/mptensor/package.py (hunk 1 above)
Comment (kuramoto-fj) [reply var/spack/repos/builtin/packages/mptensor/package.py]: Yes, I reverted them.

File: var/spack/repos/builtin/packages/mptensor/package.py (hunk 2 above)
Comment (kuramoto-fj) [reply var/spack/repos/builtin/packages/mptensor/package.py]: Sorry , I reverted them too.

File: var/spack/repos/builtin/packages/mptensor/package.py [hunk 3]
```diff
@@ -53,3 +55,42 @@ def cmake_args(self):
...
+
+            with working_dir(join_path(self.install_test_root, "tests"), create=False):
+                make("LDFLAGS={0}".format(math_libs.ld_flags))
+
+                mpirun = self.spec["mpi"].prefix.bin.mpirun
+                mpiexec = Executable(mpirun)
+                mpiexec("-n", "1", "tensor_test.out")
+
+                """ Test of mptensor has checker
+                    and checker is abort when check detect any errors. """
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mptensor/package.py]: Please use comments, not string literals:
```suggestion
                # Test of mptensor has checker
                # and checker is abort when check detect any errors.
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Another minor change, otherwise LGTM. Asking for a second reviewer since test support is somewhat new (so checking if there are common idioms already)

File: var/spack/repos/builtin/packages/mptensor/package.py (hunk 3 above)
Comment (kuramoto-fj) [reply var/spack/repos/builtin/packages/mptensor/package.py]: Thank you for your suggestions.
I fixed it as you suggested.

**Review by alalazo — APPROVED** [review]

**Comment by kuramoto-fj** [comment]
I got it.
Thank you for your reviewing and requesting to other reviewer.

**Comment by kuramoto-fj** [comment]
Could you give this PR a review, adamjstewart-san?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**